Google.Api.Generator/Generation/MethodDetails.cs

```diff
@@ -269,6 +269,12 @@ private static MethodDetails DetectPagination(ServiceDetails svc, MethodDescript
                 return null;
             }
 
+            if (pageSizeCandidate.Name == "max_results" && pageSizeCandidate.MessageType.FullName == "google.protobuf.UInt32Value")
```

Comment:
`pageSizeCandidate.MessageType` will return null if `pageSizeCandidate` isn't a message-oriented field.

But I don't see how this made it through `FindPageSizeCandidate` in the first place, as that uses IsValidPageSizeCandidate, which should return null because the field isn't an Int32 field or a UInt32 field.

I would have thought we'd *want* to use this field though - it's still a page size field really. We just need to work out how to do so properly, right? So is this just a temporary change until we work out how to use the field properly?

Reply:
:+1: , fixed the nullref.

- If FindPageSizeCandidate cannot find the candidate that passes all criteria it returns one that has just the correct name. Then if all other candidates are present we throw. This is what happens here. 
- It's not a `page_size` field, it's a `max_results` field (this is why this error never happened before). So the error happens for a non-AIP LRO for libraries with handwritten veneers. I don't think updating the generator to handle this case is a priority.